plutus-core/plutus-ir/src/PlutusIR/Transform/LetFloatIn.hs

```diff
@@ -125,12 +141,16 @@ whose LHS is used more than once, and we descend in all other cases;
 -}
 
 -- | The uniques of all used variables in a term.
-type Uniques = Set PLC.TermUnique
+type Uniques = (Set PLC.TermUnique, Set PLC.TypeUnique)
+
+noCommonUniqs :: Uniques -> Uniques -> Bool
```

Comment:
Tempted to say we should pull in `lattices` and define `disjoint a b = a /\ b == bottom` and use that, but probably it's overkill.

Reply:
I like lattices.

Reply:
I forgot about `Set.disjoint`. That should be enough.